pyproject.toml

```diff
@@ -16,7 +16,7 @@ classifiers = [
 requires-python = ">=3.7"
 dependencies = [
     "flask>=2.2.5",
-    "sqlalchemy>=1.4.18",
+    "sqlalchemy>=2.0.16",
```

Comment:
That's reasonable, I just got bit indirectly by Pillow 10.0 change in another repo, so it seems like the responsible thing for a package is to explicitly check compatibility with new major versions.

I've just changed it to:
"sqlalchemy>=2.0.16,sqlalchemy<2.1",

Reply:
Added note to the changelog

Reply:
Libraries must _never_ set upper bounds on dependencies, it makes it very difficult for installers to resolve dependency trees. See https://iscinumpy.dev/post/bound-version-constraints/

Reply:
Ah okay, I've removed the version cap, still in process of reading the post, thanks for letting me know.

Reply:
Update: Going to revert this, and make the tests run in both SQLAlchemy 1.4 and SQLAlchemy 2 (see my note in conftest.py)

Reply:
"this" includes the `>=2` requirement I assume? (i think it'll be a LONG time until everyone upgraded to SA2, for small apps it might be easy but for larger one, in particular if there's no perfect test coverage, it's not likely to happen quickly, so keeping support for SA 1.4 until upstream discontinues it is much better IMHO)

Reply:
Update: I started along the path of removing the upper bound of 2, but realized that would require a lot more checks inside our codebase, since we wouldn't even be able to import the new classes without a version check.

After conferring with @davidism, we decided to stick with the upper bound of >=2. 

There seems to be good adoption of SA2, and this version still supports all the 1.4 syntax, so it shouldn't break 1.4 code (if all goes as planned). 

We wouldn't remove 1.4 support entirely until a major bump of this package, and we'd put deprecation warnings in before that happens.

SA version download charts: https://www.pepy.tech/projects/sqlalchemy?versions=2.*&versions=1.*